### PR TITLE
[Improvement](sort) Free sort blocks if this block is exhausted

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/vec/common/sort/sorter.cpp [hunk 1]
```diff
@@ -59,48 +59,45 @@ namespace doris::vectorized {
...
+    std::vector<std::shared_ptr<MergeSortCursorImpl>> empty_cursors(0);
+    std::vector<std::shared_ptr<Block>> empty_blocks(0);
     sorted_blocks_.swap(empty_blocks);
     unsorted_block_ = Block::create_unique(unsorted_block_->clone_empty());
     in_mem_sorted_bocks_size_ = 0;
 }
 
-Status MergeSorterState::add_sorted_block(Block& block) {
-    auto rows = block.rows();
+Status MergeSorterState::add_sorted_block(std::shared_ptr<Block> block) {
```
Comment (HappenLee) [inline be/src/vec/common/sort/sorter.cpp:69]: The func seems always return OK, should be void

File: be/src/vec/core/sort_cursor.h [hunk 2]
```diff
@@ -130,37 +131,30 @@ struct MergeSortCursorImpl {
...
+            : block(Block::create_shared()), desc(desc_), sort_columns_size(desc.size()) {}
     bool empty() const { return rows == 0; }
 
     /// Set the cursor to the beginning of the new block.
-    void reset(Block& block) {
-        all_columns.clear();
+    void reset() {
         sort_columns.clear();
 
-        auto columns = block.get_columns_and_convert();
```
Comment (HappenLee) [inline be/src/vec/core/sort_cursor.h:147]: here do the const convert? can be delete safe?

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**